delete.go

```diff
@@ -0,0 +1,14 @@
+package influxdb
+
+import "context"
+
+// Predicate is something that can match on a series key.
+type Predicate interface {
+	Matches(key []byte) bool
+	Marshal() ([]byte, error)
+}
+
+// DeleteService will delete a bucket from the range and predict.
```

Comment:
Hey @goller, @e-dard 
tsm1 has the interface of Predicate, which imports influxdb. So the  as long as the DeleteService is in the influxdb layer it won't really work. We could move the DeleteService to the storage layer, I feel it's more like an anti-pattern thing. Since similar service like WriteService is in the domain level. What do you guys think?